integrations/stripe.ts

```diff
@@ -1,15 +1,96 @@
 import Stripe from "stripe"
 import { str, envsafe } from "envsafe"
+import { User } from "db"
 
 export const env = envsafe({
   DOMAIN: str(),
   STRIPE_WEBHOOK_SECRET: str(),
   STRIPE_SECRET_KEY: str(),
 })
 
-export * from "stripe"
+// https://stackoverflow.com/questions/59903956/how-to-use-stripe-types-in-typescript
 
-// Centrally configure an instance of the Stripe client
-export default new Stripe(env.STRIPE_SECRET_KEY, {
+const getStripeSecretKey = (): string => {
+  const secretKey = env.STRIPE_SECRET_KEY
+  if (!secretKey) {
+    throw new Error("Stripe secret key not found in variable 'env.STRIPE_SECRET_KEY'")
+  }
+  return secretKey
+}
+
+const stripe = new Stripe(getStripeSecretKey(), {
   apiVersion: "2020-08-27",
+  typescript: true,
+  maxNetworkRetries: 2,
 })
+
+export const getPrice = async (id: string, expand?: ["product"]) => {
+  const price = await stripe.prices
+    .retrieve(id, { expand })
+    .then((price) => price)
+    .catch((err) => {
+      console.log(err)
+    })
+  if (!price) {
+    // handle case when price is void
+  } else {
+    // Perform Additional checks here using price's data and throw any errors:
+    // such as checking if price.active is false
+  }
+  return price
+}
+
+export const getProduct = async (id: string) => {
+  const product = await stripe.products
+    .retrieve(id)
+    .then((product) => product)
```

Comment:
This isn't doing anything, it's just taking a product and returning a product. Not sure why.

Reply:
The question's more about what's the point of these wrappers. Previously, from another part of the codebase, you would do:

```js
const product = await stripe.products.retrieve(id);
```

With the wrapper, you would do

```js
const product = await getProduct(id);
```

The wrapper isn't actually making this any easier, or encapsulating anything. It's adding code in one place and not removing it anywhere.

Reply:
That would be for common error handling logic across all calls, and to allow for any modifications to the retrieve logic. One of the things I had in mind, but haven't implemented as yet, was a filter to throw an error if we tried to retrieve an inactive product/ price. (i.e., the product is retrieved successfully, but it has "active: false" in it)

I also don't think I coded the error logic right, stripe has this

`//
// An error from the Stripe API or an otheriwse asynchronous error
// will be available as the first argument of any Stripe method's callback:
// E.g. stripe.customers.create({...}, function(err, result) {});
//
// Or in the form of a rejected promise.
// E.g. stripe.customers.create({...}).then(
//        function(result) {},
//        function(err) {}
//      );`

so the integration should use that format instead

Reply:
Unless there's something that clearly will use this abstraction, I would default to [yagni](https://www.martinfowler.com/bliki/Yagni.html) here. Right now these methods are just indirection without functionality.

Reply:
Huh, that's interesting haha, thanks for sharing. 

That makes sense, I doubt any codebase will have more than a handful of calls to stuff like `await stripe.products.retrieve(id);` so error handling can be done at all of those locations for now instead of at these methods